### PR TITLE
Add transfer flow transitions

### PR DESCRIPTION
**Pull request by davelange**
Add transitions/animations to elements in the transfer flow. Does not include the bg grid, that will be in the next PR.
 - Add a `Transition` component, for better control of elements that are animated
 - Add transitions to global CSS, as this is easier to understand than doing it in tailwind classes.
 - Some fixes in the `TextInput` component
 - Add a dark mode to the transaction status indicator

### CODE REVIEW

**Comment by davelange** [comment]
Looks like there's still a CI failure

File: package-lock.json
Comment (juliangruber) [inline package-lock.json:1]: The package lock changed a lot but the package.json didn't 🤔

**Review by juliangruber — APPROVED** [review]
Fantastic work 👏 I have only the one question about `package-lock.json`

File: package-lock.json
Comment (davelange) [reply package-lock.json:1]: https://github.com/filecoin-station/desktop/actions/runs/9387558869/job/25879602948
The build was failing in CI for linux and windows with `Error: Cannot find module @rollup/rollup-linux-x64-gnu. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828). ` In the [linked issue](https://github.com/npm/cli/issues/4828) the suggested fix is to delete package.lock and node_modules and reinstall everything to make sure all the peer deps are included

**Pull request merged**